homeassistant/components/binary_sensor/deconz.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for deCONZ binary sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    DECONZ_DATA, DOMAIN)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.core import callback
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup binary sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_BINARY_SENSOR
+    sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
+        if sensor.type in DECONZ_BINARY_SENSOR:
+            async_add_devices([DeconzBinarySensor(sensor)], True)
+
+
+class DeconzBinarySensor(BinarySensorDevice):
+    """Representation of a binary sensor."""
+
+    def __init__(self, sensor):
+        """Setup sensor and add update callback to get data from websocket."""
+        self._sensor = sensor
+        self._sensor.register_callback(self._update_callback)
+
+    @callback
+    def _update_callback(self, reason):
+        """Update the sensor's state, if reason is that state is updated."""
+        if reason['state']:
+            self.async_schedule_update_ha_state()
```

Comment:
Don't call async methods from a sync context (I doubt deconz schedules your callback on the event loop?)

Reply:
All callbacks within entities originates from asyncio transport data_received